apps/src/music/player/MusicPlayer.ts

```diff
@@ -416,6 +431,9 @@ export default class MusicPlayer {
 
       const folder: SoundFolder | null = library.getFolderForFolderId(kit);
 
+      const length = patternEvent.value.length || DEFAULT_PATTERN_LENGTH;
+      const eventsLength = length * 16;
```

Comment:
nit: maybe `numSixteenths` or `numTicks`? though `ticks` is a bit of misnomer and now might be extra confusing since we use it in the AI code 🙂